### PR TITLE
ci: Fix quick-jobs GHC version to 9.6

### PR DESCRIPTION
**Pull request by alt-romes**
This should fix the CI quick jobs that have been failing recently. Could you please have a look @Mikolaj

### CODE REVIEW

File: .github/workflows/quick-jobs.yml [hunk 1]
```diff
@@ -25,15 +25,13 @@ jobs:
...
-      #     ghcup --version
-      #     ghcup config set cache true
-      #     ghcup install ghc recommended
-      #     ghcup set ghc recommended
+      - name: ghcup
+        run: |
+          ghcup --version
+          ghcup config set cache true
+          ghcup install ghc 9.8.2
+          ghcup set ghc 9.8.2
```
Comment (mpickering) [inline .github/workflows/quick-jobs.yml:34]:
```suggestion
          ghcup install ghc 9.6.5
          ghcup set ghc 9.6.5
```

**Review by Mikolaj — APPROVED** [review]
Thank you. Let's expedite this fix, since CI is broken currently due to this problem.

**Comment by Mikolaj** [comment]
I mean, after CI passes. :)

**Review by geekosaur — APPROVED** [review]

**Comment by ulysses4ever** [comment]
The other quick jobs are failing now. For the same reason, presumably. A more future-proof solution would be to make them all work with 9.10.

**Comment by ulysses4ever** [comment]
I pushed a version with the fix propagated to other quick jobs. It brings horrible duplication. At least, I created a variable for the version of GHC.

**Comment by Mikolaj** [comment]
@mergify backport 3.12

**Pull request merged**